cogs/help.py

```diff
@@ -1,63 +1,185 @@
-from nextcord import ui, ButtonStyle, ChannelType, Interaction, MessageType
+from typing import Dict, Optional
+
+from nextcord import Button, ButtonStyle, ChannelType, Colour, Embed, Interaction, Member, MessageType, Thread, ui
 from nextcord.ext import commands
-import nextcord
+
+HELP_CHANNEL_ID: int = 881965127031722004
+HELP_LOGS_CHANNEL_ID: int = 883035085434142781
+HELPER_ROLE_ID: int = 882192899519954944
+
+
+class HelpButton(ui.Button["HelpView"]):
+    def __init__(self, help_type: str, *, style: ButtonStyle, custom_id: str):
+        super().__init__(label = f"{help_type} help", style = style, custom_id = custom_id)
+        self._help_type = help_type
+
+    async def create_help_thread(self, interaction: Interaction) -> None:
+        name = "Discord.py libraries" if self._help_type == "D.py libs" else self._help_type
+
+        thread = await interaction.channel.create_thread(
+            name = f"{name} help ({interaction.user})",
+            type = ChannelType.public_thread
+        )
+        # don't need this since the ping already adds them.
```

Comment:
Dead code should be removed

Reply:
Left it for review purposes for if you wanted it to be added again idk, will remove it.